drgn_tools/nfs_tools.py

```diff
@@ -0,0 +1,931 @@
+
+    print(
+        "---- <nfs_client: 0x%x>  cl_minorversion: %d  cl_ipaddr: %s"
+        % (nfs_client.value_(), nfs_client.cl_minorversion, local_ip)
+    )
+    print(
+        "     cl_hostname: %s  cl_addr: %d.%d.%d.%d [remote]"
+        % (
+            server_name,
+            remote_ip[0] & 0xFF,
+            remote_ip[1] & 0xFF,
+            remote_ip[2] & 0xFF,
+            remote_ip[3] & 0xFF,
+        )
+    )
+
+    print(
+        "     cl_flags: 0x%x %s"
+        % (nfs_client.cl_flags, nfs_client_flag_str(nfs_client.cl_flags))
+    )
+    if is_UEK5(prog):
+        print("     cl_count: %d" % (nfs_client.cl_count.counter))
+    else:
+        print("     cl_count: %d" % (nfs_client.cl_count.refs.counter))
+
+    if nfs_client.cl_rpcclient.cl_auth:
+        RpcAuthFlavors = enum_type_to_class(
+            prog.type("enum rpc_auth_flavors"), "RpcAuthFlavors"
+        )
+        enum_name = enum_name_get(
+            RpcAuthFlavors,
+            nfs_client.cl_rpcclient.cl_auth.au_flavor,
+            "UNKNOWN",
+        )
+    else:
+        enum_name = "NULL"
+    print(
+        "     cl_rpcclient: 0x%x  cl_auth.au_flavor: %d (%s)"
+        % (
+            nfs_client.cl_rpcclient.value_(),
+            nfs_client.cl_rpcclient.cl_auth.au_flavor,
+            enum_name,
+        )
+    )
+
+    nfs_servers = nfs_server_list(nfs_client)
+    print("     nfs_server: %d" % len(nfs_servers))
+    for ns in nfs_servers:
+        mount_src = ""
+        mount_dst = ""
+        mntp = get_mount(prog, ns.super)
+        if mntp is not None:
+            mount_src = escape_ascii_string(mntp.mnt_devname.string_())
+            mount_dst = escape_ascii_string(
+                d_path(mntp.mnt.address_of_(), mntp.mnt.mnt_root)
+            )
+
+        print(
+            "          <nfs_server: 0x%x>  %s %s"
+            % (ns.value_(), mount_src, mount_dst)
+        )
+        print(
+            "          flags: 0x%x %s"
+            % (ns.flags, nfs_server_flag_str(ns.flags))
+        )
+        print(
+            "          super: 0x%x  s_count: %d  s_active: %d"
+            % (ns.super.value_(), ns.super.s_count, ns.super.s_active.counter)
+        )
+        print(
+            "          client: 0x%x  nfs_client: 0x%x"
+            % (ns.client.value_(), ns.nfs_client.value_())
+        )
+        display_nfs_delegations(ns)
+        print("")
+
+
+def show_nfs_clients(prog) -> None:
+    """
+    Display summary information of the NFS layer on the NFS client side.
+    The information includes major data structures such as nfs_client,
+    nfs_server and corresponding RPC transport.
+
+    :param prog: drgn pogram
+    :returns: None
+    """
+
+    clients = nfs_client_list(prog)
+    print(
+        "============================== NFS_CLIENT ================================"
+    )
+    for clnt in clients:
+        show_nfs_client(prog, clnt)
+
+
+def nfsshow(prog: drgn.Program, max_tasks: int = 10) -> None:
```

Comment:
It looks like this is the main entry point where you'd expect most people to call for summary NFS info. I'd recommend that you create a corelens module to call this function. It would look something like this:
```python
class NfsShow(CorelensModule):
    """Print summary information on NFS client side"""

    name = "nfsshow"

    # there's no point in running unless the nfs module is loaded
    skip_unless_have_kmod = "nfs"

    # we need sunrpc, and all nfs related debuginfo loaded
    debuginfo_kmods = ["sunrpc", "nfs", "nfs*"]

    def add_args(self, parser: argparse.ArgumentParser) -> None:
        parser.add_argument(
            "--max-tasks",
            type=int,
            default=10,
            help="list at most <number> rpc_tasks",
        )

    def run(self, prog: Program, args: argparse.Namespace) -> None:
        nfsshow(prog, max_tasks=args.max_tasks)
```